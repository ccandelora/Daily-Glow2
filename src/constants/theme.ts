export const COLORS = {
  // Primary colors for emotions (keeping these for emotional context)
  primary: {
    red: '#FF6B6B',     // Warm red for negative emotions
    green: '#00C853',   // Vibrant green for positive emotions
    blue: '#2979FF',    // Bright blue for calm emotions
    yellow: '#FFD600',  // Bright yellow for energetic emotions
  },
  
  // UI colors updated for dark theme with glowing accents
  ui: {
    background: 'rgba(28, 14, 45, 0.95)',  // Dark purple background
    card: 'rgba(38, 20, 60, 0.85)',        // Slightly lighter purple for cards
    text: '#FFFFFF',                        // White text for contrast
    textSecondary: 'rgba(255, 255, 255, 0.7)', // Semi-transparent white
    border: 'rgba(82, 67, 194, 0.3)',      // Subtle purple border
    disabled: 'rgba(82, 67, 194, 0.2)',    // Dimmed purple for disabled state
    accent: '#4169E1',                     // Royal blue for accents
    highlight: 'rgba(65, 105, 225, 0.15)', // Soft blue highlight
  },

  // Status colors with glow effects
  status: {
    success: '#00E676',
    error: '#FF5252',
    warning: '#FFD600',
    info: '#4169E1',
  },

  // Gradient colors for glow effects
  gradient: {
    start: 'rgba(65, 105, 225, 0.2)',  // Royal blue glow
    middle: 'rgba(147, 112, 219, 0.15)', // Purple glow
    end: 'rgba(28, 14, 45, 0.9)',     // Dark purple base
  },
};

export const FONTS = {
  sizes: {
    xs: 12,
    sm: 14,
    md: 16,
    lg: 22,
    xl: 28,
    xxl: 36,
    xxxl: 44,
  },
  weights: {
    regular: '400' as const,
    medium: '500' as const,
    semibold: '600' as const,
    bold: '700' as const,
  },
  // Adding font families for more sophistication
  families: {
    heading: 'System',
    body: 'System',
  }
};

export const SPACING = {
  xs: 4,
  sm: 8,
  md: 16,
  lg: 24,
  xl: 32,
  xxl: 40,
};

export const BORDER_RADIUS = {
  sm: 4,
  md: 8,
  lg: 16,
  xl: 24,
  circle: 999,
};

export const TIME_PERIODS = {
  MORNING: {
    label: 'Morning',
    range: { start: 5, end: 11 }, // 5:00 AM - 11:59 AM
    greeting: 'Good morning',
    icon: 'üåÖ'
  },
  AFTERNOON: {
    label: 'Afternoon',
    range: { start: 12, end: 16 }, // 12:00 PM - 4:59 PM
    greeting: 'Good afternoon',
    icon: '‚òÄÔ∏è'
  },
  EVENING: {
    label: 'Evening',
    range: { start: 17, end: 4 }, // 5:00 PM - 4:59 AM
    greeting: 'Good evening',
    icon: 'ÔøΩÔøΩ'
  }
} as const;

export default {
  COLORS,
  FONTS,
  SPACING,
  BORDER_RADIUS,
  TIME_PERIODS,
}; 