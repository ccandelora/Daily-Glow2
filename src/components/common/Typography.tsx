import React from 'react';
import { Text, StyleSheet, TextStyle, TextProps } from 'react-native';
import theme from '@/constants/theme';

type GlowIntensity = 'none' | 'soft' | 'medium' | 'strong';

interface TypographyProps extends TextProps {
  variant?: 'h1' | 'h2' | 'h3' | 'body' | 'caption' | 'button';
  color?: string;
  style?: TextStyle;
  glow?: GlowIntensity;
}

export const Typography: React.FC<TypographyProps> = ({
  children,
  variant = 'body',
  color,
  style,
  glow = 'none',
  ...props
}) => {
  const getGlowStyle = (intensity: GlowIntensity) => {
    if (intensity === 'none') return null;
    const glowStyles = {
      soft: styles.glowSoft,
      medium: styles.glowMedium,
      strong: styles.glowStrong,
    };
    return glowStyles[intensity];
  };

  return (
    <Text
      style={[
        styles.base,
        styles[variant],
        color && { color },
        getGlowStyle(glow),
        style,
      ]}
      {...props}
    >
      {children}
    </Text>
  );
};

const styles = StyleSheet.create({
  base: {
    color: theme.COLORS.ui.text,
    fontSize: theme.FONTS.sizes.md,
    fontFamily: theme.FONTS.families.body,
  },
  h1: {
    fontSize: theme.FONTS.sizes.xxxl,
    fontWeight: theme.FONTS.weights.bold,
    fontFamily: theme.FONTS.families.heading,
    color: theme.COLORS.ui.text,
    textShadowColor: theme.COLORS.ui.accent,
    textShadowOffset: { width: 0, height: 1 },
    textShadowRadius: 4,
  },
  h2: {
    fontSize: theme.FONTS.sizes.xxl,
    fontWeight: theme.FONTS.weights.bold,
    fontFamily: theme.FONTS.families.heading,
    color: theme.COLORS.ui.text,
    textShadowColor: theme.COLORS.ui.accent,
    textShadowOffset: { width: 0, height: 1 },
    textShadowRadius: 3,
  },
  h3: {
    fontSize: theme.FONTS.sizes.xl,
    fontWeight: theme.FONTS.weights.semibold,
    fontFamily: theme.FONTS.families.heading,
    color: theme.COLORS.ui.text,
    textShadowColor: theme.COLORS.ui.accent,
    textShadowOffset: { width: 0, height: 1 },
    textShadowRadius: 2,
  },
  body: {
    fontSize: theme.FONTS.sizes.md,
    fontWeight: theme.FONTS.weights.regular,
    lineHeight: 24,
    color: theme.COLORS.ui.textSecondary,
  },
  caption: {
    fontSize: theme.FONTS.sizes.sm,
    fontWeight: theme.FONTS.weights.regular,
    color: theme.COLORS.ui.textSecondary,
  },
  button: {
    fontSize: theme.FONTS.sizes.md,
    fontWeight: theme.FONTS.weights.semibold,
    textAlign: 'center',
  },
  glowSoft: {
    textShadowColor: theme.COLORS.ui.accent,
    textShadowOffset: { width: 0, height: 0 },
    textShadowRadius: 4,
  },
  glowMedium: {
    textShadowColor: theme.COLORS.ui.accent,
    textShadowOffset: { width: 0, height: 0 },
    textShadowRadius: 8,
  },
  glowStrong: {
    textShadowColor: theme.COLORS.ui.accent,
    textShadowOffset: { width: 0, height: 0 },
    textShadowRadius: 12,
  },
}); 