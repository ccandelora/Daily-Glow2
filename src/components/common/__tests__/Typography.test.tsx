import React from 'react';
import { render } from '@testing-library/react-native';
import { Typography } from '../Typography';
import { TextStyle } from 'react-native';
import theme from '@/constants/theme';

describe('Typography Component', () => {
  test('renders correctly with default props', () => {
    const { getByText, toJSON } = render(
      <Typography>Default Text</Typography>
    );
    
    expect(getByText('Default Text')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot('default-typography');
  });
  
  test('applies correct styles for each variant', () => {
    // Test each variant
    const variants = ['h1', 'h2', 'h3', 'body', 'caption', 'button'] as const;
    
    variants.forEach(variant => {
      const { getByTestId } = render(
        <Typography variant={variant} testID={`text-${variant}`}>
          {variant} Text
        </Typography>
      );
      
      const textElement = getByTestId(`text-${variant}`);
      
      // Just verify the style array contains objects
      const styleArray = textElement.props.style;
      expect(Array.isArray(styleArray)).toBe(true);
      expect(styleArray.length).toBeGreaterThan(1);
    });
  });
  
  test('applies custom color when provided', () => {
    const customColor = '#FF5733';
    const { getByTestId } = render(
      <Typography color={customColor} testID="colored-text">
        Colored Text
      </Typography>
    );
    
    const textElement = getByTestId('colored-text');
    
    // Check that color is applied
    expect(textElement.props.style.some((style: any) => 
      style && style.color === customColor
    )).toBe(true);
  });
  
  test('applies custom styles when provided', () => {
    const customStyle: TextStyle = { 
      letterSpacing: 2,
      textDecorationLine: 'underline',
    };
    
    const { getByTestId } = render(
      <Typography style={customStyle} testID="custom-styled-text">
        Custom Styled Text
      </Typography>
    );
    
    const textElement = getByTestId('custom-styled-text');
    
    // Check custom style is applied (should be last in the style array)
    const styleArray = textElement.props.style;
    expect(styleArray[styleArray.length - 1]).toMatchObject(customStyle);
  });
  
  test('applies glow effect styles', () => {
    // Test each glow intensity
    const intensities = ['soft', 'medium', 'strong'] as const;
    
    intensities.forEach(intensity => {
      const { getByTestId } = render(
        <Typography glow={intensity} testID={`text-glow-${intensity}`}>
          {intensity} Glow Text
        </Typography>
      );
      
      const textElement = getByTestId(`text-glow-${intensity}`);
      
      // Check that appropriate glow style is applied
      const styleArray = textElement.props.style;
      const hasGlowStyle = styleArray.some((style: any) => 
        style && style.textShadowRadius && 
        (
          (intensity === 'soft' && style.textShadowRadius === 4) ||
          (intensity === 'medium' && style.textShadowRadius === 8) ||
          (intensity === 'strong' && style.textShadowRadius === 12)
        )
      );
      
      expect(hasGlowStyle).toBe(true);
    });
  });
  
  test('passes additional props to Text component', () => {
    const numberOfLines = 2;
    const { getByTestId } = render(
      <Typography 
        testID="text-with-props" 
        numberOfLines={numberOfLines}
        accessibilityLabel="Accessible text"
      >
        Text with additional props
      </Typography>
    );
    
    const textElement = getByTestId('text-with-props');
    
    expect(textElement.props.numberOfLines).toBe(numberOfLines);
    expect(textElement.props.accessibilityLabel).toBe('Accessible text');
  });
}); 