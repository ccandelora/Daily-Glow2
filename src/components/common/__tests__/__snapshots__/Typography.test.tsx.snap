// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Component renders correctly with default props: default-typography 1`] = `
<Text
  style={
    [
      {
        "color": "#FFFFFF",
        "fontFamily": "System",
        "fontSize": 16,
      },
      {
        "color": "rgba(255, 255, 255, 0.7)",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 24,
      },
      undefined,
      null,
      undefined,
    ]
  }
>
  Default Text
</Text>
`;
